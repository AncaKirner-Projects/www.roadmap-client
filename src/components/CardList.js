import React, { Component } from 'react';
import CardElement from './CardElement';

class CardList extends Component {
  render() {
    return (
      <div>
        {
          this.props.products.map((product) =>
            <CardElement key={product.id} product={product} />)
        }
      </div>
    );
  }
}

export default CardList;
