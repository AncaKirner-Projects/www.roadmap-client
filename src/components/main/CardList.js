import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CardElement from './CardElement';

const CardList = (props) => {
  const { products } = props;
  return (
    <React.Fragment>
      {
        products && products.map(product => <CardElement key={product.id} product={product} />)
      }
    </React.Fragment>
  );
};

CardList.propTypes = {
  products: PropTypes.array
};

CardList.defaultProps = {
  products: []
};

const mapStateToProps = (state) => {
  const products = (state.categories && state.categories.selectedProducts)
    ? state.categories.selectedProducts
    : state.products.uniq;

  return {
    products
  };
};

export default connect(mapStateToProps)(CardList);
