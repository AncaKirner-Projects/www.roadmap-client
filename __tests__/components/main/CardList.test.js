import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import CardList from '../../../src/components/main/CardList';

const mockStore = configureMockStore();

describe('<CardList />', () => {
  it('renders a card list', () => {
    const store = mockStore({
      categories: {
      },
      products: {
        uniq: [
          {
            id: 1,
            prod_name: 'Test Product',
            prod_description: 'Product description',
            price: 23.6
          }
        ]
      }
    });

    const wrapper = shallow(<CardList store={store} />);
    const component = wrapper.dive();

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders a card list when categories selected', () => {
    const store = mockStore({
      categories: {
        selectedProducts: [
          {
            id: 1,
            prod_name: 'Test Product',
            prod_description: 'Product description',
            price: 23.6
          }
        ]
      }
    });
    const wrapper = shallow(<CardList store={store} />);
    const component = wrapper.dive();

    expect(toJson(component)).toMatchSnapshot();
  });
});
