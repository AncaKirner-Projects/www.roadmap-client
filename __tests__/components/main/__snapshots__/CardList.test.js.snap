// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> renders a card list 1`] = `
<Fragment>
  <CardElement
    key="1"
    product={
      Object {
        "id": 1,
        "price": 23.6,
        "prod_description": "Product description",
        "prod_name": "Test Product",
      }
    }
  />
</Fragment>
`;

exports[`<CardList /> renders a card list when categories selected 1`] = `
<Fragment>
  <CardElement
    key="1"
    product={
      Object {
        "id": 1,
        "price": 23.6,
        "prod_description": "Product description",
        "prod_name": "Test Product",
      }
    }
  />
</Fragment>
`;
